tests/integration/targets/mso_backup/tasks/main.yml

```diff
@@ -43,540 +44,774 @@
+      ignore_errors: yes
+
+    - name: Move a ansibleBackup3 from local location to remote
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup3
+        remote_location: ansible_test
+        remote_path: test
+        state: move
+      when: version.current.version is version('3', '>=')
+      register: move_ansibleBackup3
+      ignore_errors: yes
+
+    - name: Verify move function of backup
+      assert:
+        that:
+        - move_backup_cm is changed
+        - move_backup_nm is changed
+        - move_non_existent_backup.msg is match ("Backup 'non_existent_backup' does not exist")
+        - move_ansibleBackup3.msg is match ("Multiple backups with same name found. Existing backups with similar names{{':'}} ansibleBackup3_[0-9]*, ansibleBackup3_[0-9]*")
+      when: version.current.version is version('3', '>=')
+
+
+# Create Backups only for NDO version >= 3.2
+- name: Create Backups only for NDO version >= 3.2
+  when: version.current.version is version('3.2', '>=')
+  block:
+    - name: Create remote ansibleBackup3 in normal mode
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup3
+        description: via Ansible
+        location_type: remote
+        remote_location: ansible_test
+        remote_path: /tmp
+        state: present
+      register: nm_add_ansibleBackup3
+
+    - name: Query ansibleBackup3 to check if it exists
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup3
+        state: query
+      register: check_ansibleBackup3
+      until:
+      - check_ansibleBackup3.current is defined
+      - check_ansibleBackup3.current[0] is defined
+      - check_ansibleBackup3.current[0].status.statusType == "success"
+      retries: 50
+      delay: 10
+
+    - name: Verify nm_add_ansibleBackup3
+      assert:
+        that:
+        - nm_add_ansibleBackup3 is changed
+        - check_ansibleBackup3.current[0].backupEntry.metadata.name is match ("ansibleBackup3_[0-9a-zA-Z]*")
+
+    - name: Create remote ansibleBackup3 in normal mode again
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup3
+        description: via Ansible
+        location_type: remote
+        remote_location: ansible_test
+        remote_path: /tmp
+        state: present
+      register: nm_add_ansibleBackup3_again
+
+    - name: Query ansibleBackup3 to check if it exists again
+      mso_backup:
+        <<: *mso_info
+        backup: ansibleBackup3
+        state: query
+      register: check_ansibleBackup3_again
+      until:
+      - check_ansibleBackup3_again.current is defined
+      - check_ansibleBackup3_again.current[0] is defined
+      - check_ansibleBackup3_again.current[0].status.statusType == "success"
+      retries: 50
+      delay: 10
+
+    - name: Verify nm_add_ansibleBackup3_again
+      assert:
+        that:
+        - check_ansibleBackup3_again.current[0].backupEntry.metadata.name is match ("ansibleBackup3_[0-9a-zA-Z]*")
```

Comment:
Remove this block as block in L 202 is already covering most of this and everything when adding again.